sunpy/coordinates/transformations.py

```diff
@@ -56,6 +56,16 @@ def _carrington_offset(obstime):
     from .ephemeris import get_sun_L0
     return get_sun_L0(obstime)
 
+
+def _observers_are_equal(obs_1, obs_2):
+    if obs_1 == obs_2:
```

Comment:
Should *this* check the observers are not strings? or do we do it somewhere else in here? I just realised that this will pass if both the observers are "earth" which probably isnt a good thing.

Reply:
Yeah I'm not really sure...

Reply:
There is already a check in a few places to make sure that the observer is an instance of `BaseCoordinateFrame`. Maybe we should add an additional check in here? You can't really compare two observers if they are only strings right?

Reply:
no, just string matching isnt sufficient for the spirit of this function I dont think.

Reply:
I agree.  earth is not always earth.  The two observers are the same if they have the same location.  Not sure that the two observer times have to be the same.

Reply:
So then what about [this test](https://github.com/sunpy/sunpy/blob/master/sunpy/coordinates/tests/test_transformations.py#L75)? HPC->HPC but the frames are not defined by an observer or obstime

Reply:
BTW this one fails when I remove the `obs_1==obs_2` check and add a restriction that both are `BaseCoordinateFrame` instances.

Reply:
hmmm thats an interesting one. I think that call should probably error, if we dont know the obstime we should probably not pretend we can do any transforms at all. even ones that could just be null like that test.